policies/local.go

```diff
@@ -0,0 +1,147 @@
+//  Copyright 2019 Google Inc. All Rights Reserved.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+
+package policies
+
+import (
+	"bytes"
+	"encoding/json"
+
+	agentendpointpb "github.com/GoogleCloudPlatform/osconfig/_internal/gapi-cloud-osconfig-go/google.golang.org/genproto/googleapis/cloud/osconfig/agentendpoint/v1alpha1"
+	"github.com/golang/protobuf/jsonpb"
+	"github.com/golang/protobuf/proto"
+)
+
+// localConfig represents the structure of the config to the JSON parser.
+//
+// The types of members of the struct are wrappers for protobufs and delegate
+// the parsing to jsonpb lib via their UnmarshalJSON implementations.
+type localConfig struct {
```

Comment:
Is there a reason to do it this way vs just using jsonpb.Unmarshal to localConfig directly? That would eliminate the needs for all the UnmarshalJSON methods and eliminate the intermediate structs.

Reply:
That was my initial idea. Changed after discussion with ryanwe@. The rationale is that the local software config does not need the source fields, therefore the JSON can be simplified.

Reply:
I don't think I'm full understanding, I'm saying create this stuct:

```
type localConfig struct {
  Packages            []agentendpointpb.Package
  PackageRepositories []agentendpointpb.PackageRepository
  SoftwareRecipes     []agentendpointpb.SoftwareRecipe
}
```

then just:

var lc localConfig
jsonpb.Unmarshal(data, &lc)

Reply:
Okay, I think I misunderstood your question. So, for jsonpb to unmarshal into localConfig, it would have to be a proto, otherwise it won't compile: 
 ```
cannot use &lc (type *localConfig) as type proto.Message in argument to jsonpb.Unmarshal: *localConfig does not implement proto.Message (missing ProtoMessage method)
```
So we'd have to either use the full response proto (which we decided against) or define localConfig as a proto, which seemed a bit too heavyweight just to have JSON parsing.